tianshou/data/batch.py

```diff
@@ -639,7 +601,40 @@ def __cat(self,
 
     def cat_(self,
              batches: Union['Batch', List[Union[dict, 'Batch']]]) -> None:
-        return self.__cat(batches)
+        """Concatenate a list of (or one) :class:`~tianshou.data.Batch` objects
+        into current batch.
+
+        If x=Batch(a=Batch(a=np.random.randn(3, 4)), b=np.random.randn(3, 4)),
+        y = Batch(a=Batch(a=Batch()), b=np.random.randn(3, 4)), if we want to
```

Comment:
I don'k like much this example since it implies that `a=Batch(a=Batch())` has to be specified.

Reply:
I think it is okay. It is a corner case that requires some thinking. 

``x=Batch(a=Batch(a=np.random.randn(3, 4)), b=np.random.randn(3, 4), c=np.random.randn(3, 2))`` is straightforward and actually previous code works for it, by treating 'c' as partial keys.

So we had better keep it to let future maintainers (including us) understand the necessity of ``batch_holder.__cat(v, lens=lens)``.

Reply:
As you wish.